File: backend/Services/Parsers/GlobleParser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class GlobleParser : BasicScoreResultParser
    {
        private ILogger<GlobleParser> _logger;

        public GlobleParser(ILogger<GlobleParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => true;
        public override string GameName => "Globle";
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($"🌎.*?🌍[\\s\n\r]+[^=]*= (?<{ScoreGroup}>\\d+)");
        protected override string? ExtraContent => null;
        public override string Url => "https://globle-game.com";
        protected override string GetCleanResult(string result, Match parserResults)
        {
            return result
                .Replace("#globle", string.Empty)
                .Replace(Url, string.Empty)
                .Trim();
        }
    }
}

File: lambdas/src/parsers/globle-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class GlobleParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Globle';
  readonly helpText = null;
  protected readonly extraContent = null;
  readonly url = 'https://globle-game.com';
  
  // Matches "🌎 Sep 4, 2024 🌍\n...\n🟨🟥🟩 = 3"
  protected readonly parser = new RegExp(
    `🌎.*?🌍[\\s\\n\\r]+[^=]*= (?<${BasicScoreResultParser.SCORE_GROUP}>\\d+)`
  );
  
  protected getCleanResult(result: string, match: RegExpMatchArray): string {
    return result
      .replace('#globle', '')
      .replace(this.url ?? '', '')
      .trim();
  }
}
